addons/hr/views/res_users.xml

```diff
@@ -205,12 +205,27 @@
             <field name="inherit_id" ref="base.view_users_simple_form"/>
             <field name="arch" type="xml">
                 <xpath expr="//field[@name='mobile']" position="after">
-                    <field name="create_employee" force_save="1" string="Create Employee" invisible="not context.get('allow_create_employee', True)" attrs="{'invisible': [('id', '>', 0)]}"/>
                     <field name="create_employee_id" force_save="1" invisible="1"/>
+                    <field name="create_employee" force_save="1" string="Create Employee" invisible="not context.get('allow_create_employee', True)" attrs="{'invisible': [('create_employee_id', '>', 0)]}"/>
                 </xpath>
             </field>
         </record>
 
+        <record id="view_users_simple_form_inherit_hr_save_and_close" model="ir.ui.view">
+            <field name="name">view.users.simple.form.inherit.hr.save.and.close</field>
```

Comment:
The inherit is not needed when the inherit mode is primary
https://www.odoo.com/documentation/15.0/contributing/development/coding_guidelines.html#inheriting-xml